lib/rules/no-duplicate-attributes.js

```diff
@@ -37,27 +37,48 @@ function getName (attribute) {
  * @returns {Object} AST event handlers.
  */
 function create (context) {
-  const names = new Set()
+  const options = context.options[0] || {}
+  const allowCoexistStyle = Boolean(options.allowCoexistStyle !== false)
+  const allowCoexistClass = Boolean(options.allowCoexistClass !== false)
+
+  const directiveNames = new Set()
+  const attributeNames = new Set()
+
+  function isDuplicate (name) {
+    if (allowCoexistStyle && name === 'style') {
+      return directiveNames.has(name)
+    }
+    if (allowCoexistClass && name === 'class') {
+      return directiveNames.has(name)
```

Comment:
I'd like to see the test of the following case:

```html
<div :class="a" class="b"></div>
```

I guess that the `class="b"` is warned even if `allowCoexistClass` is true.